### PR TITLE
JOINDIN-717 #close

### PR DESCRIPTION
**Pull request by cal-tec**
If you comment on an event with a rating of 0 then the comment gets created successfully with an HTTP 201 but looking at the code the error that is displayed if a comment is rated > 5 is 'The field "rating" must be a number (1-5)'
This implies that a comment with a 0 rating should return the same error and not a 201 Created

Due to an irc discussion it became apparent that the constraints on the rating field should only be applied when a rating was allowable (if not previously rated and not an event host). It was however indicated that the rating field should be required in all situations to make the api easier to use & document.

This by default solves a fairly major (and i believe unidentified) bug in that if allowable a rating could be submitted of any value of 5 or less (including negative values).

### CODE REVIEW

**Comment by heiglandreas** [comment]
Just to clarify: 

> Due to an irc discussion it became apparent that the constraints on the rating field should only be applied when a rating was allowable (if not previously rated and not an **event host**). It was however indicated that the rating field should be required in all situations to make the api easier to use & document.

Should that not be the speaker of the talk and not the event host?

**Comment by cal-tec** [comment]
No it is supposed to read like that, this is related to event comments only.

**Comment by heiglandreas** [comment]
Sorry, TL/DR... YOu'Re absolutely right!

File: src/controllers/Event_commentsController.php [hunk 1]
```diff
@@ -74,8 +74,8 @@ public function createComment($request, $db)
         $rating = $request->getParameter('rating', false);
         if (false === $rating) {
             throw new Exception('The field "rating" is required', 400);
-        } elseif (false === is_numeric($rating) || $rating > 5) {
-            throw new Exception('The field "rating" must be a number (1-5)', 400);
+        } elseif (false === is_numeric($rating)) {
```
Comment (heiglandreas) [inline src/controllers/Event_commentsController.php]: Can we remove the ```elseif``` here? There's an exception thrown before so the method is left there already. 

I'd prefer two separate ```if```-statements without the else.

**Pull request merged**